RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1001,30 +1303,37 @@ class DeepTauId : public deep_tau::DeepTauBase {
     return predictions;
   }
 
-  void getPredictionsV1(const TauType& tau,
-                        const pat::ElectronCollection& electrons,
-                        const pat::MuonCollection& muons,
-                        std::vector<tensorflow::Tensor>& pred_vector) {
-    const tensorflow::Tensor& inputs = createInputsV1<dnn_inputs_2017v1>(tau, electrons, muons);
+  template <typename CandidateCastType, typename TauCastType>
+  void getPredictionsV1(const TauType::const_reference& tau,
+                        size_t tau_index,
```

Comment:
Here, and in all similar methods: Is there a reason to have tau_index as non-const argument? At least I did not notice any. If not, please turn into const.

Reply:
There is not, this has been changed.